packages/syft/src/syft/oblv/deployment.py

```diff
@@ -0,0 +1,128 @@
+from .oblv_proxy import create_oblv_key_pair
+from .oblv_proxy import get_oblv_public_key
+
+SUPPORTED_REGION_LIST = ["us-east-1", "us-west-2", "eu-central-1", "eu-west-2"]
+
+SUPPORTED_INFRA = ["c5.xlarge", "m5.xlarge", "r5.xlarge", "c5.2xlarge", "m5.2xlarge"]
+
+
+def create_deployment(
+    domain_clients: list,
+    deployment_name: Optional[str] = None,
+    key_name: str = "",
+    oblv_client: Optional[OblvClient] = None,
+    infra: str = INFRA,
+    region: str = REGION,
+) -> str:
+    """Creates a new deployment with predefined codebase
+    Args:
+        client : Oblivious Client.
+        domain_clients: List of domain_clients.
+        deployment_name: Unique name for the deployment.
+        key_name: User's key to be used for deployment creation.
+        infra: Represent the AWS infrastructure to be used. Default is "m5.2xlarge". The available options are\n
+                - "c5.xlarge" {'CPU':4, 'RAM':8, 'Total/hr':0.68}\n
+                - "m5.xlarge" {'CPU':4, 'RAM':16, 'Total/hr':0.768}\n
+                - "r5.xlarge" {'CPU':4, 'RAM':32, 'Total/hr':1.008}\n
+                - "c5.2xlarge" {'CPU':8, 'RAM':16, 'Total/hr':1.36}\n
+                - "m5.2xlarge" {'CPU':8, 'RAM':32, 'Total/hr':1.536}\n
+                As of now, PySyft only works with RAM >= 32
+        region: AWS Region to be deployed in. Default is "us-east-1". The available options are \n
+                - "us-east-1" : "US East (N. Virginia)",\n
+                - "us-west-2" : "US West (Oregon)",\n
+                - "eu-central-1" : "Europe (Frankfurt)",\n
+                - "eu-west-2" : "Europe (London)"
+
+    Returns:
+        resp: Deployment Client Object
+    """
+    if not oblv_client:
+        oblv_client = login()
+    if deployment_name is None:
+        deployment_name = input("Kindly provide deployment name")
+    if key_name == "":
+        key_name = input("Please provide your key name")
+
+    while not SUPPORTED_INFRA.__contains__(infra):
+        infra = input(
+            "Provide infra from one of the following - {}".format(
+                ",".join(SUPPORTED_INFRA)
+            )
+        )
+    while not SUPPORTED_REGION_LIST.__contains__(region):
+        region = input(
+            "Provide region from one of the following - {}".format(
+                ",".join(SUPPORTED_REGION_LIST)
+            )
+        )
+    try:
+        user_public_key = get_oblv_public_key(key_name)
+    except FileNotFoundError:
+        user_public_key = create_oblv_key_pair(key_name)
+        print(
+            bcolors.green(bcolors.bold("Created"))
+            + f" a new public/private key pair with key_name: {key_name}"
+        )
+    except Exception as e:
+        raise Exception(e)
+    build_args = {
+        "auth": {},
+        "users": {"domain": [], "user": []},
+        "additional_args": {},
+        "infra_reqs": infra,
+        "runtime_args": "",
+    }
+    users = []
+    runtime_args = []
+    for k in domain_clients:
```

Comment:
OUGHTTO: Could we name this variable to something else `k` doesn't indicate much?